pkg/kubernetes/nodestore_test.go

```diff
@@ -0,0 +1,427 @@
+package kubernetes
+
+import (
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"testing"
+)
+
+var (
+	nodeUID1 = generateRandomNodeUUID()
+	nodeUID2 = generateRandomNodeUUID()
+)
+
+func Test_SetOrUpdateNode(t *testing.T) {
+	tests := []struct {
+		name          string
+		node          v1.Node
+		existingNodes *NodeStore
```

Comment:
Is there any reason why `existingNodes` and `wantNodes` are a `NodeStore`? I think that we could simplify this by using `[]metav1.ObjectMeta` instead.